server/server.go

```diff
@@ -330,6 +339,7 @@ func New(
 		config:            *config,
 		monitorsChan:      make(chan struct{}),
 		defaultIDMappings: idMappings,
+		pullCache:         make(map[string]*pullResult),
```

Comment:
It's unclear to me where this pullCache is used

Reply:
In `PullImage(...)`. I added more verbose comments (also to the internal types) to make the code more approachable.